### ISSUE TITLE
feat: add attention bias parameter to FlashAttention and FlashAttentionBackward in IEngine

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet.Tensors#12]
## Summary

The `IEngine.FlashAttention` and `IEngine.FlashAttentionBackward` methods currently do not accept an additive attention bias parameter. This forces consumers like `FlashAttentionLayer` in AiDotNet to fall back to a manual (non-flash) attention path whenever attention biases are needed, losing the performance benefits of flash attention.

## Current Signature

```csharp
Tensor<T> FlashAttention(
    Tensor<T> queries,
    Tensor<T> keys,
    Tensor<T> values,
    double? scale,
    bool useCausalMask,
    out Tensor<T>? softmaxStats);

void FlashAttentionBackward(
    Tensor<T> outputGradient,
    Tensor<T> queries,
    Tensor<T> keys,
    Tensor<T> values,
    Tensor<T> output,
    Tensor<T> softmaxStats,
    double scale,
    bool useCausalMask,
    out Tensor<T> queryGradient,
    out Tensor<T> keyGradient,
    out Tensor<T> valueGradient);
```

## Proposed Change

Add an optional `attentionBias` parameter to both methods:

```csharp
Tensor<T> FlashAttention(
    Tensor<T> queries,
    Tensor<T> keys,
    Tensor<T> values,
    double? scale,
    bool useCausalMask,
    out Tensor<T>? softmaxStats,
    Tensor<T>? attentionBias = null);  // NEW: additive bias applied before softmax

void FlashAttentionBackward(
    Tensor<T> outputGradient,
    Tensor<T> queries,
    Tensor<T> keys,
    Tensor<T> values,
    Tensor<T> output,
    Tensor<T> softmaxStats,
    double scale,
    bool useCausalMask,
    out Tensor<T> queryGradient,
    out Tensor<T> keyGradient,
    out Tensor<T> valueGradient,
    Tensor<T>? attentionBias = null);  // NEW: same bias for backward gradient computation
```

## Use Cases

1. **ALiBi (Attention with Linear Biases)**: Per-head linear distance biases injected before softmax. Currently requires a full manual attention fallback in `FlashAttentionLayer`, defeating the purpose of flash attention.

2. **Relative position biases**: T5-style relative position biases, Transformer-XL biases, and other pre-softmax additive biases all need the same capability.

3. **Custom attention masking**: Additive masks (e.g., padding masks as large negative values) could be passed as bias instead of requiring separate masking logic.

## Context

This was identified while implementing ALiBi support in AiDotNet PR #830. The workaround is a manual attention path (`ComputeAttentionWithALiBi`) that computes `QK^T`, adds the ALiBi bias, applies softmax, and multiplies by V - but this loses flash attention's memory and speed optimizations.

Reference: Lines 288-305 in `src/NeuralNetworks/Attention/FlashAttentionLayer.cs`

## Implementation Notes

- The bias tensor shape should be `[numHeads, queryLen, keyLen]` or `[batch, numHeads, queryLen, keyLen]`
- When `attentionBias` is null, behavior is identical to current (backward compatible)
- For CPU engine: add bias to attention scores after `QK^T * scale` and before softmax
- For GPU engines: pass bias to the underlying CUDA flash attention kernel (FlashAttention-2 supports this natively via `attn_bias` parameter)
- Backward pass: bias gradient is simply the softmax gradient (since bias is additive)
